src/In.ProjectEKA.FHIRHip/DataFlow/Collect.cs

```diff
@@ -0,0 +1,123 @@
+            var careContextReferences = patientData.Keys.ToList();
+            foreach (var careContextReference in careContextReferences)
+                foreach (var result in patientData.GetOrDefault(careContextReference))
+                {
+                    Log.Information($"Returning file: {result}");
+                    var fjp = new FhirJsonParser();
+                    bundles.Add(new CareBundle(careContextReference, fjp.Parse<Bundle>(result)));
+                }
+
+            var entries = new Entries(bundles);
+            return Option.Some(entries);
+        }
+
+        private async Task<Dictionary<string, List<string>>> FindPatientsData(DataRequest request)
+        {
+            // LogDataRequest(request);
+            
+            var patientReferenceNumber = request.CareContexts.First().PatientReference;
+            var careContexts = request.CareContexts.Select(careContext => careContext.CareContextReference).ToList();
+            var dataResponse = await GetPatientsData(new NetworkDataRequest(patientReferenceNumber,
+                                                             careContexts,
+                                                             request.DateRange,
+                                                             request.HiType));
+            var structuredData = new Dictionary<string, List<string>>();
+            return dataResponse.Map(content =>
+            {
+                foreach (var result in content.Results)
+                {
+                    if (structuredData.ContainsKey(result.CareContext))
+                    {
+                        structuredData[result.CareContext].Add(result.Data);
+                    }
+                    else
+                    {
+                        structuredData.Add(result.CareContext, new List<string> {result.Data});
+                    }
+                }
+                return structuredData;
+            }).ValueOr(structuredData);
+        }
+
+        private async Task<Option<NetworkDataResponse>> GetPatientsData(NetworkDataRequest networkDataRequest)
+        {
+            try
+            {
+                var json = JsonConvert.SerializeObject(networkDataRequest, new JsonSerializerSettings
+                {
+                    NullValueHandling = NullValueHandling.Ignore,
+                    ContractResolver = new DefaultContractResolver
+                    {
+                        NamingStrategy = new CamelCaseNamingStrategy()
+                    }
+                });
+                var httpRequestMessage = new HttpRequestMessage(HttpMethod.Post, new Uri($"{HipSystemDataUrl}"))
+                {
+                    Content = new StringContent(json, Encoding.UTF8, MediaTypeNames.Application.Json)
+                };
+                var response = await HttpClient.SendAsync(httpRequestMessage).ConfigureAwait(false);
+                if (!response.IsSuccessStatusCode)
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=6044012)